docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,343 @@
+call to \<reconnect()>. This code is safe, however: \<reconnect()>
+ensures that \<sock> is closed before re-assigning it.
+
+Support for \<@CreatesMustCallFor> can be disabled by passing the \<-AnoCreatesMustCallFor> command-line
+argument to the checker. In this mode, \<@CreatesMustCallFor> annotations on methods are ignored at call-sites,
+and re-assigning a non-final, owning field is always an error.
+
+Sections~\ref{resource-leak-createsmustcallfor-callsite}
+and~\ref{resource-leak-createsmustcallfor-declaration}
+explain how the Resource Leak Checker verifies uses and declarations of
+methods annotated with \<@CreatesMustCallFor>.
+
+
+\subsectionAndLabel{Requirements at a call site of a \<@CreatesMustCallFor> method}{resource-leak-createsmustcallfor-callsite}
+
+At a call site to a method annotated as
+\<@CreatesMustCallFor(>\emph{expr}\<)>, the Resource Leak Checker:
+\begin{enumerate}
+\item
+  Treats any existing \<@MustCall> obligations of \emph{expr} as \emph{satisfied},
+\item
+  Creates a fresh obligation to check, as if \emph{expr} was a newly-allocated local
+  variable.
+\item
+  Requires that the expression corresponding to \emph{expr} (that is, \emph{expr}
+  viewpoint-adapted to the method call site) is owned; that is, it is
+  annotated or defaulted as \<@Owning>.  Otherwise, the checker
+  will issue a \<reset.not.owning> error at the call-site. You can avoid this
+  error by extracting the relevant expression into a new local variable (because
+  locals are \<@Owning> by default).
+\end{enumerate}
+
+Treating the obligation before the call as satisfied is sound: the
+checker creates a new obligation for calls to \<@CreatesMustCallFor> methods,
+and the Must Call Checker (\chapterpageref{must-call-checker}) ensures the
+\<@MustCall> type for the target will have a \emph{superset} of any methods
+present before the call. Intuitively, calling an \<@CreatesMustCallFor> method
+``resets'' the obligations of the target, so whether they were satisfied before
+the call or not is irrelevant.
+
+If an \<@CreatesMustCallFor>
+method is invoked within a method that has an \<@CreatesMustCallFor> annotation
+with the same target---imposing the obligation on its caller---then
+the newly-created obligation is treated as satisfied immediately (because it is instead imposed
+on the callee of the method being checked).
+
+
+\subsectionAndLabel{Requirements at a declaration of a \<@CreatesMustCallFor> method}{resource-leak-createsmustcallfor-declaration}
+
+The Resource Leak Checker enforces two rules to ensure that
+re-assignments to non-final, owning fields like \<sock> in the example
+above are sound:
+\begin{itemize}
+\item any method that re-assigns a non-final, owning field of an object
+  must be annotated with a \<@CreatesMustCallFor> annotation
+  that targets that object.
+\item when a non-final, owning field $f$ is re-assigned at statement $s$,
+  its inferred \<@MustCall> obligation must be contained in its \<@CalledMethods>
```

Comment:
Why is it important that it is "inferred"?  It cannot be declared?
Is this fact after the assignment?
Is this a fact about `s` or about the right-hand side?
Please clarify.

Reply:
This is a fancy way of saying that `f`'s type immediately before the assignment must have satisfied its must-call requirements. I've rephrased the entire requirement to be more plain English and less paper text.